### ISSUE TITLE
[Feature Request] Binding gamepad buttons to Citra hotkeys

### ISSUE THREAD

**Issue by HenrikoMagnifico** [citra-emu/citra#5885]
This is a feature request for the option to bind gamepad inputs to hotkeys within Citra. The reasons as to why this would be useful are several;

- It would let the user change between the top and bottom screens using their controller only, with no need to access a keyboard or mouse during gameplay.
- Creating and loading save states without a keyboard.
- Pausing / resuming emulation at any point without a keyboard.
- Taking screenshots during quickly during hectic gameplay.

Et cetera... The options are nearly endless, and would drastically improve the user experience as well as accessibility options.

**Comment by SoloBackup** [comment]
This would be a spectacular function, as it exists in other emulators. Hope they implement this feature

**Comment by Daisouji** [comment]
I agree this is common and good feature for a lot of emulators.

**Comment by ZeptoBST** [comment]
+1

**Comment by Penixlast14** [comment]
+1

**Comment by Jeimuzu277** [comment]
I couldn't agree more.  Especially for screen swapping.

I think screen layout toggling would be another great reason for cutscenes that use both screens.

**Comment by Holorodney** [comment]
+1 for sure

**Comment by askotx** [comment]
A workaround, that could solve our issue, is using joytokey software, like I did. In windows, not tested on linux, I created a Citra profile for my xbox 360 controller, then I set uncommon used keys for citra shorcuts:
F1 - Exit for L3 button
F11: Fullscreen for L2 button
F10: Swap screen for Trigger button R
F9: Change layout screen for Trigger button L

Then associate this profile when citra-qt.exe is running, in Settings. Just tinker a little bit with joytokey program

**Comment by Daisouji** [comment]
> A workaround, that could solve our issue, is using joytokey software, like I did. In windows, not tested on linux, I created a Citra profile for my xbox 360 controller, then I set uncommon used keys for citra shorcuts: F1 - Exit for L3 button F11: Fullscreen for L2 button F10: Swap screen for Trigger button R F9: Change layout screen for Trigger button L
> 
> Then associate this profile when citra-qt.exe is running, in Settings. Just tinker a little bit with joytokey program

LOL, we know this already. We want it added to the actual program.

**Comment by askotx** [comment]
As I stated is a workaround, nobody said it's an issue solver. Anyway, if it solves some part of the problem, what's the matter?
I'm not seeing any other one offering some kind of solution, also I want this feature added too, It will be offer a better user experience for front end integration.

**Comment by TomTurbine** [comment]
+1 Can definitely agree with this. Have a PS5 controller hooked up and would love to bind the Touchpad, PS Button and Mic buttons to hotkeys.

**Comment by emufa** [comment]
This would also be huge, would make the emulator a 10/10 for me.

**Comment by Anayka-Ryu** [comment]
Hey I couldn undestanding almost nothing

**Comment by Anayka-Ryu** [comment]
I would to need a translator on Spanish i would using the cheats of shinies Money Items and more but What happened the black Screens on postgame or alola photoclub?

**Comment by Anayka-Ryu** [comment]
10/10 What means that?

**Comment by Anayka-Ryu** [comment]
But i just playing on Android

**Comment by mark4409** [comment]
In particular, I think the Android version would really benefit from this as they don't really have hotkeys as such, instead relying on bringing up an in game menu and then toggling screens. This really isn't a very fluid operation, especially when using a controller. I know the android developer @t895 is currently rolling out numerous improvements in stages (which have been really useful) so wondered if something like this was in his plans for the near future?

**Comment by TomTurbine** [comment]
It's been a little over a year since I last posted for this option. Any word on the progress with it? I thought this would be one of the easier things to add.

Being able to bind Exit, Pause, and Swap screens to my PS Logo, Microphone, and Touchscreen buttons on my PS5 controller would be a nice plus.

**Comment by t895** [comment]
It was added! It's available in the latest nightly build right now!

**Comment by askotx** [comment]
But it was just for Android. 
Windows and Linux are not done, techically is not complete. OP never specified any OS.

**Comment by t895** [comment]
Oh sorry about that. It's just surprising to me that Android got a feature first for once 😅

**Comment by TomTurbine** [comment]
Yeah, I don't use my phone for gaming outside of the restroom. I use my PC for gaming.

And find it weird how it can support any button on my controller for any input on the system but not the hotkeys.

**Comment by zhen-huan-hu** [comment]
It would be essential for Steam Deck (or any other handheld PC). yuzu supports controller hotkeys.

**Comment by lasers** [comment]
If somebody can look at making https://github.com/AntiMicroX/antimicrox work on Windows and macOS too, then this feature would be possible for not just Citra, but other emulators and/or games. I already use Citra with this on Linux via keyboard hotkeys.
